internal/fs/inode/dir.go

```diff
@@ -850,18 +852,19 @@ func (d *dirInode) DeleteChildDir(
 			Generation: 0, // Delete the latest version of object named after dir.
 		})
 
-	if err != nil {
-		err = fmt.Errorf("DeleteObject: %w", err)
-		return
-	}
-
 	if d.bucket.BucketType() == gcs.Hierarchical {
-		// Delete Folder deletes folder (in case of Hierarchical Bucket).
+		// Ignoring delete object error here, as in case hns there is no way of knowing
+		// if underlying placeholder object exists or not
+		// The DeleteFolder operation handles removing empty folders.
 		err = d.bucket.DeleteFolder(ctx, childName.GcsObjectName())
+		if err != nil {
+			return fmt.Errorf("DeleteFolder: %w", err)
```

Comment:
We (me and @ankitaluthra1) discussed this and reached to conclusion for  Hierarchical bucket We concluded that we can safely ignore errors from DeleteObject since we also call DeleteFolder, which will return any err if occur.